ContactControles.js

```diff
@@ -0,0 +1,142 @@
+var app = angular.module('myApp', ["ngRoute"]);
+
+app.config(function($routeProvider,$locationProvider) {
+     $routeProvider.when('/', {
+               templateUrl : 'home.html',
+               controller: 'contactsCtrl'
+     }).when('/editContact/:contact_index', {
+               templateUrl : 'addContact.html',
+               controller: 'editCtrl'
+     }).when('/addContact', {
+               templateUrl : 'addContact.html',
+               controller: 'addCtrl'
+                         }).otherwise({
+         redirectTo: '/'
+                         });
+ });
+
+app.controller('contactsCtrl', function($scope, ContactService, $http, $location, $window) {
+  $scope.contacts = ContactService.getContacts(); //Yehia: Tabs Tabs
+  $scope.goToHome =function(path) {
+    $location.path(path);
+  }
+
+  $scope.goToEdit =function(contact_index) {
+    if(contact_index =="new") {
+      $location.url("/addContact");
+    }
+    else {
+      $location.url("/editContact/"+contact_index);
+    }
+  }
+
+  $scope.deleteContact = function(id) {
+    var deleteUser = $window.confirm('Are you sure you want to delete the Ad?');
+    if(deleteUser) {
+      ContactService.deleteContact(id);
+      alert("The Contact was deleted");
+    }
+    else  {
+      alert("The Contact wasn't deleted yet");
+    }
+  }
+
+});
+
+app.controller('editCtrl', function($scope,ContactService,$routeParams,$location){

```

Comment:
Naming convention: my suggestion is to rename this controller to `editContactCtrl` or `contactDetailsCtrl` to better understand the controller's purpose.